faraday-multipart.gemspec

```diff
@@ -31,5 +31,5 @@ Gem::Specification.new do |spec|
 
   spec.required_ruby_version = '>= 2.4', '< 4'
 
-  spec.add_dependency 'multipart-post', '>= 1.2', '< 3'
+  spec.add_dependency 'multipart-post', '~> 2'
```

Comment:
Did you mean to declare a dependency on any future version of `multipart-post`?  Presumably 3.0, 4.0, 17.0, etc will likely have breaking changes, but this version constraint is essentially `>= 2.0`.  If you want to to depend on any 2.x version this should probably be `~> 2.0`.

Reply:
I'd advise to use `~> 2.0`.

Reply:
Thanks! https://github.com/lostisland/faraday-multipart/commit/a3609a6921cd032252a9cee5d1388982e54bc045 now uses that.

Reply:
This change is unnecessary as `~> 2` is equivalent to `~> 2.0`:

```
irb(main):001:0> Gem::Requirement.new('~> 2').satisfied_by? Gem::Version.new("2.0")
=> true
irb(main):002:0> Gem::Requirement.new('~> 2').satisfied_by? Gem::Version.new("2.2")
=> true
irb(main):003:0> Gem::Requirement.new('~> 2').satisfied_by? Gem::Version.new("1")
=> false
irb(main):004:0> Gem::Requirement.new('~> 2').satisfied_by? Gem::Version.new("3.0")
=> false
irb(main):005:0> Gem::Requirement.new('~> 2').satisfied_by? Gem::Version.new("4.0")
=> false
```

From Rubygem docs https://guides.rubygems.org/patterns/:


> It’s also important to know that if you specify a major version only, like this:
>
> `spec.add_runtime_dependency 'library', '~> 2'`
>
> It will only use the latest version from the 2.x series – so 2.3.0 – and not 3.0.0. This behaviour may surprise some people, but automatically allowing any major version past version 2 is more surprising behaviour.

Reply:
Thanks, @jsvd, for details!

(The change in the gemspec is now "the less surprising one".)

Reply:
@jsvd thanks for teaching me something new! I didn't realize rubygems had a special case for a single digit...

Reply:
I didn't know it either! Thanks.

Reply:
TIL! Thank you @jsvd 🙏